src/pa02/test_pa02.py

```diff
@@ -0,0 +1,148 @@
+    assert b.goal == 90
+
+
+def test_goal_reached():
+    """Checking if goal_reached works"""
+    b = cs.Board()
+    assert b.goal_reached(90) is True
+    assert b.goal_reached(13) is False
+    assert b.goal_reached(100) is True
+
+
+def test_position_adjustment():
+    """Checking if position_adjustment works"""
+    b = cs.Board()
+    assert b.position_adjustment(24) == -19
+    assert b.position_adjustment(68) == 17
+    assert b.position_adjustment(7) == 0
+
+
+def test_player_move():
+    """Checking that player is initially at 0, and not in the same position
+    after move is called"""
+    b = cs.Board()
+    p = cs.Player(b)
+    assert p.position == 0
+    p.position = 5
+    p.move()
+    assert p.position != 5
+
+
+def test_resilient_player():
+    """Checking if ResilientPlayer behaves as desired"""
+    random.seed(999)
+    b = cs.Board()
+    p = cs.ResilientPlayer(b)
+    p.position = 36
+    p.move()
+    random.seed(999)
+    p.move()
+    assert p.position == 37
+
+
+def test_lazy_player():
+    """Checking if LazyPlayer behaves as desired"""
+    random.seed(999)
+    b = cs.Board()
+    p = cs.LazyPlayer(b)
+    p.position = 2
+    p.move()
+    random.seed(999)
+    p.move()
+    assert p.position == 15
+
+
+def test_types():
+    """Checking if returned player types are correct"""
+    b = cs.Board()
+    rp = cs.ResilientPlayer(b)
+    assert type(rp).__name__ == 'ResilientPlayer'
+    lp = cs.LazyPlayer(b)
+    assert type(lp).__name__ == 'LazyPlayer'
+
+
+def test_random_order():
+    """Adds test to see if elements in a list is same sorted as unsorted"""
+    list_of_players = [cs.Player, cs.LazyPlayer, cs.ResilientPlayer]
+    b = cs.Board()
+    s1 = cs.Simulation(list_of_players, b, 999)
+    s2 = cs.Simulation(list_of_players, b, 999, False)
+    dict1 = s1.players_per_type()
+    dict2 = s2.players_per_type()
+    assert dict1 == dict2
+
+
+def test_single_game():
+    """Checking if single_game behaves correctly"""
+    s = cs.Simulation(player_field=[cs.Player, cs.ResilientPlayer,
+                                    cs.LazyPlayer], seed=999)
+    n, winner_type = s.single_game()
+    assert type(n) == int
+    assert (winner_type == 'Player' or winner_type == 'ResilientPlayer' or
+            winner_type == 'LazyPlayer')
+
+
+def test_run_simulation():
+    """Checking that results list changes after run_simulation is called"""
+    s = cs.Simulation(player_field=[cs.Player, cs.ResilientPlayer,
+                                    cs.LazyPlayer], seed=999)
+    s.run_simulation(2)
+    first_results = s.results.copy()
+    s.run_simulation()
+    later_results = s.results
+    assert first_results != later_results
+
+
+def test_get_results():
+    """Checking if get_results works"""
+    s = cs.Simulation(player_field=[cs.Player, cs.ResilientPlayer,
+                                    cs.LazyPlayer], seed=999)
+    s.run_simulation(2)
+    assert s.get_results() == s.results
```

Comment:
Not really convincing here as this is what's in your code.

Reply:
Which makes this test case has no values.
Testing get_results can just test for:
- Return values: list of tuple
- 1st item in tuple: an int that is greater than 0
- 2nd item in the tuple: a str that is in 'Player', 'ResilientPlayer' or 'LazyPlayer'

Reply:
If possible, you can make a custom board that have ladder from 1 to 90, 2 to 90, 3 to 90, 4 to 90, 5 to 90, 6 to 90.
Then, any player that plays first wins. And you can explicitly specify the expected return value for get_results.